core/azure-core/src/main/java/com/azure/core/implementation/configuration/Configuration.java

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.configuration;
+
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Contains configuration information that is used during construction of service clients.
+ */
+public final class Configuration {
+
+    /**
+     * Empty Configuration object used to opt out of using global configurations when constructing clients.
+     */
+    public static final Configuration NONE = new Configuration();
+
+    private Map<String, String> configurations = new HashMap<>();
+
+    Configuration() {
```

Comment:
There would be no harm in having this be public, for the case where the user wants an empty configuration that they then modify. Currently I think they would need to do `Configuration.NONE.clone().put(...).put(...)`

Reply:
Made empty constructor public